From 4e41f90e07c8203d096c77b1a3c1e162dc92fea8 Mon Sep 17 00:00:00 2001
From: Roy Pledge <roy.pledge@nxp.com>
Date: Mon, 6 May 2019 11:18:57 -0400
Subject: [PATCH 509/741] sdk_qbman: Fix error in IP revision comparison

commit 097f25152b7918034db9e1ab6033f817a042a3b9 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

The comparison for QMAN_REV31 was incorrect as it
would always fail due to the wrong mask.

This fixes the following error in newer GCC versions:
"error: bitwise comparison always evaluates to false
	[-Werror=tautological-compare]"

Signed-off-by: Roy Pledge <roy.pledge@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/staging/fsl_qbman/qman_config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/fsl_qbman/qman_config.c b/drivers/staging/fsl_qbman/qman_config.c
index 040631354df5..529f840117ea 100644
--- a/drivers/staging/fsl_qbman/qman_config.c
+++ b/drivers/staging/fsl_qbman/qman_config.c
@@ -812,7 +812,7 @@ int qman_set_sdest(u16 channel, unsigned int cpu_idx)
 
 	if (!qman_have_ccsr())
 		return -ENODEV;
-	if ((qman_ip_rev & 0xFF00) == QMAN_REV31) {
+	if ((qman_ip_rev & 0xFFFF) == QMAN_REV31) {
 		/* LS1043A - only one L2 cache */
 		cpu_idx = 0;
 	}
-- 
2.17.1

